basis/common-router/src/main/java/com/softgroup/common/router/api/factories/AbstractHandlerFactory.java

```diff
@@ -0,0 +1,29 @@
+package com.softgroup.common.router.api.factories;
+
+import com.softgroup.common.protocol.Request;
+import com.softgroup.common.router.api.Handler;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Created by Виктор on 03.03.2017.
+ */
+public abstract class AbstractHandlerFactory<T extends Handler> {
+    protected Map<String, T> handlerMap;
```

Comment:
I have an  AbstractHandlerFactory with abstract method getHandler(Request<?> msg) and two factories TypeHandlerFactory and CommandHandlerFactory, that extend AbstractHandlerFactory. They have different implementations of method getHandler(). That's why Map<String, T> handlerMap is protected . I need to have access to the map in a superclass. If it would be better to implement factories without AbstractHandlerFactory, I can change my code